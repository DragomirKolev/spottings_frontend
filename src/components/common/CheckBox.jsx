import React from 'react';
import { createUseStyles } from 'react-jss';

import { GLOBAL_SPACING } from '../../styles/common';

const useStyles = createUseStyles({
  container: {
    display: 'flex',
    flexDirection: 'row',
    marginTop: GLOBAL_SPACING,
    marginBottom: GLOBAL_SPACING
  },
  label: { marginRight: GLOBAL_SPACING * 2 },
});

const CheckBox = ({ label, value, onChange }) => {
  const classes = useStyles();

  return (
    <div className={classes.container}>
      <div className={classes.label}>{label} : </div>
        <input
          type="checkbox" 
          cheked={value.toString()} 
          onChange={() => onChange(!value)}
        />
    </div>
  );
}

export default CheckBox;