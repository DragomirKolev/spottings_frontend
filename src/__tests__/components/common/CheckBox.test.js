import React from 'react';
import renderer from 'react-test-renderer';

import CheckBox from '../../../components/common/CheckBox';

it('renders CheckBox', () => {
  const tree = renderer
    .create(<CheckBox 
          label={'Confident in your sighting?'}
          value={true} 
          onChange={jest.fn()}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
