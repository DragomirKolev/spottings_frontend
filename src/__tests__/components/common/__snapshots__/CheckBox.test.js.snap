// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CheckBox 1`] = `
<div
  className="container-0-2-1"
>
  <div
    className="label-0-2-2"
  >
    Confident in your sighting?
     : 
  </div>
  <input
    cheked="true"
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
